essentials-framework/Essentials Core/PepperDashEssentialsBase/Remotes/Hrxx0WirelessRemoteController.cs

```diff
@@ -0,0 +1,233 @@
+using Crestron.SimplSharp.Reflection;
+using Newtonsoft.Json;
+
+
+using PepperDash.Core;
+using PepperDash.Essentials.Core;
+using PepperDash.Essentials.Core.Config;
+using PepperDash.Essentials.Core.Bridges;
+
+namespace PepperDash.Essentials.Core
+{
+    [Description("Wrapper class for all HR-Series remotes")]
+    public class Hrxx0WirelessRemoteController : EssentialsBridgeableDevice, IHasFeedback
+    {
+        private Hr1x0WirelessRemoteBase _remote;
+
+        public FeedbackCollection<Feedback> Feedbacks { get; set; }
+
+        public CrestronCollection<Button> Buttons { get { return _remote.Button; } }
+
+        public Hrxx0WirelessRemoteController(string key, Func<DeviceConfig, Hr1x0WirelessRemoteBase> preActivationFunc, 
+            DeviceConfig config)
+            : base(key, config.Name)
+        {
+            Feedbacks = new FeedbackCollection<Feedback>();
+
+            AddPreActivationAction(() =>
+            {
+                _remote = preActivationFunc(config);
+
+                _remote.ButtonStateChange += new ButtonEventHandler(_remote_ButtonStateChange);
+
+                Feedbacks.Add(new BoolFeedback("BatteryCritical", () => _remote.BatteryCriticalFeedback.BoolValue));
+                Feedbacks.Add(new BoolFeedback("BatteryLow", () => _remote.BatteryLowFeedback.BoolValue));
+                Feedbacks.Add(new IntFeedback("BatteryVoltage", () => _remote.BatteryVoltageFeedback.UShortValue));
+
+                _remote.BaseEvent += new BaseEventHandler(_remote_BaseEvent);
+            });
+        }
+
+        void _remote_BaseEvent(GenericBase device, BaseEventArgs args)
+        {
+            if(args.EventId == Hr1x0EventIds.BatteryCriticalFeedbackEventId)
+                Feedbacks["BatteryCritical"].FireUpdate();
+            if(args.EventId == Hr1x0EventIds.BatteryLowFeedbackEventId)
+                Feedbacks["BatteryLow"].FireUpdate();
+            if(args.EventId == Hr1x0EventIds.BatteryVoltageFeedbackEventId)
+                Feedbacks["BatteryVoltage"].FireUpdate();
+        }
+
+        void _remote_ButtonStateChange(GenericBase device, ButtonEventArgs args)
+        {
+            try
+            {
+                var handler = args.Button.UserObject;
+
+                if (handler == null) return;
+
+                Debug.Console(1, this, "Executing Action: {0}", handler.ToString());
+
+                if (handler is Action<bool>)
+                {
+                    (handler as Action<bool>)(args.Button.State == eButtonState.Pressed ? true : false);
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.Console(2, this, "Error in ButtonStateChange handler: {0}", e);
+            }
+        }
+
+
+        #region Preactivation
+
+        private static Hr1x0WirelessRemoteBase GetHr1x0WirelessRemote(DeviceConfig config)
+        {
+            var props = JsonConvert.DeserializeObject<CrestronRemotePropertiesConfig>(config.Properties.ToString());

```

Comment:
Since config.Properties is a JToken here, it's easier to write something like `config.Properties.ToObject<CrestronRemotePropertiesConfig>();` instead of serializing the JToken back to a string just to deserialize it again. Less steps, quicker operation.